spring-graphql/src/main/java/org/springframework/graphql/execution/DefaultGraphQlSourceBuilder.java

```diff
@@ -193,6 +194,9 @@ private TypeDefinitionRegistry parseSchemaResource(Resource schemaResource) {
 		catch (IOException ex) {
 			throw new IllegalArgumentException("Failed to load schema resource: " + schemaResource);
 		}
+		catch (SchemaProblem ex) {
+			throw new InvalidSchemaResourceException(schemaResource.getDescription(), ex);
```

Comment:
The reason to create  InvalidSchemaResourceException is that I think the 'wrong schema config' not match the semantic of IllegalArgumentException.